pkgs/top-level/all-packages.nix

```diff
@@ -22836,7 +22836,7 @@ with pkgs;
     withPerl = false;
     # We don't use `with` statement here on purpose!
     # See https://github.com/NixOS/nixpkgs/pull/10474#discussion_r42369334
-    modules = [ nginxModules.rtmp nginxModules.dav nginxModules.moreheaders ];
+    modules = [ nginxModules.brotli nginxModules.rtmp nginxModules.dav nginxModules.moreheaders ];
```

Comment:
Those changes where really dangerous for moreheaders because if you still had it in ``services.nginx.additionalModules`` then the module would be compiled twice in without any errors and only at runtime when handling they would crash. We must handle this better from now on.

My suggestions:
- assert if ``services.nginx.additionalModules`` combined with ``modules`` contains duplicates package and maybe pname wise and hard fail if so
- we must update the description of ``services.nginx.additionalModules`` because it still contain brotli.
and maybe we should only add it to ``services.nginx.additionalModules`` when brotli is enabled. Building nginx is really quick and fast.

Reply:
How to implement the first variant? I don't know how to implement module comparison mechanism.

Reply:
To be honest I am not sure but without it the migration path could become painful.

Reply:
In the release notes, I wrote that nginx is now built with brotli module by default. Maybe something needs to be added?

Reply:
I don't think a release note is enough. This can easily be missed and most people are not aware of it and could only be noticed when nginx is already crashing. Also from search.nixos.org it is also not obvious when adding new plugins that there are already some installed by default.

Reply:
Removed this changes.